spec/java/util/OptionalInt.lsl

```diff
@@ -0,0 +1,282 @@
+                val otherPresent = OptionalInt(other).present;
+                result = self.value == otherValue && self.present == otherPresent;
+            }
+            else
+            {
+                result = false;
+            }
+        }
+    }
+
+
+    fun getAsInt (): int
+    {
+        if (!present)
+            action THROW_NEW("java.util.NoSuchElementException", ["No value present"]);
+
+        result = value;
+    }
+
+
+    fun hashCode (): int
+    {
+        result = action OBJECT_HASH_CODE(value);
+    }
+
+
+    fun ifPresent (consumer: IntConsumer): void
+    {
+        required !present || (present && consumer != null);
+
+        if (present)
+        {
+            if (consumer == null)
+                self._throwNPE();
+
+            action CALL(consumer, [value]);
+        }
+    }
+
+
+    fun ifPresentOrElse (consumer: IntConsumer, emptyAction: Runnable): void
+    {
+        required !present || (present  && consumer != null);
+        required present  || (!present && emptyAction != null);
+
+        if (present)
+        {
+            if (consumer == null)
+                self._throwNPE();
+
+            action CALL(consumer, [value]);
+        }
+        else
+        {
+            if (emptyAction == null)
+                self._throwNPE();
+
+            action CALL(emptyAction, []);
+        }
+    }
+
+
+    fun isEmpty (): boolean
+    {
+        result = present == false;
+    }
+
+
+    fun isPresent (): boolean
+    {
+        result = present == true;
+    }
+
+
+    fun orElse (other: int): int
+    {
+        if (present)
+            result = value;
+        else
+            result = other;
+    }
+
+
+    fun orElseGet (supplier: IntSupplier): int
+    {
+        required supplier != null;
+
+        if (supplier == null)
+            self._throwNPE();
+
+        if (present)
+            result = value;
+        else
+            result = action CALL(supplier, []);
+    }
+
+
+    fun orElseThrow (): int
+    {
+        if (!present)
+            action THROW_NEW("java.util.NoSuchElementException", ["No value present"]);
+
+        result = value;
+    }
+
+
+    @Generic("X extends Throwable")
+    @throws(["X"], generic=true)
+    fun orElseThrow(@Generic("? extends X") exceptionSupplier: Supplier): int
+    {
+        required exceptionSupplier != null;
+
+        if (exceptionSupplier == null)
+            self._throwNPE();
+
+        if (!present)
+        {
+            val exception = action CALL(exceptionSupplier, []);
+            action THROW_VALUE(exception);
+        }
+        else
+        {
+            result = value;
+        }
+    }
+
+
+    fun stream (): IntStream
+    {
+        action NOT_IMPLEMENTED();
+
+        /*
+        if (present)
+            result = IntStream.of(value); // #problem
+        else
+            result = IntStream.empty(); // #problem
+        */
+    }
+
+
+    fun toString (): String
```

Comment:
In original method it returns: "OptionalInt["...

Reply:
Fixed